wire-grpc-server-generator/src/main/java/com/squareup/wire/kotlin/grpcserver/FileDescriptorGenerator.kt

```diff
@@ -67,19 +75,53 @@ object FileDescriptorGenerator {
     return encoded
   }
 
-  private fun addDescriptorMapProperty(builder: TypeSpec.Builder, encoded: MutableMap<String, String>) {
+  private fun addDescriptorMapProperty(
+    builder: TypeSpec.Builder,
+    encoded: MutableMap<String, String>,
+  ) {
+    val descriptorMapInitializer = CodeBlock.builder()
+    val initLines = mutableListOf<String>()
+    val encodedListChunked: List<List<Pair<String, String>>> =
+      encoded.toList().chunked(DM_CHUNK_SIZE)
+    encodedListChunked.forEachIndexed { index: Int, subEncodedList: List<Pair<String, String>> ->
+      subDescriptorMapCodeBlock(builder, subEncodedList, index)
+      if (index > 0) {
+        initLines.add(" +\n")
+      }
+      initLines.add("$DESCRIPTOR_MAP_FUNCTION_PREFIX$index()")
+    }
+
+    descriptorMapInitializer.withIndent {
+      initLines.forEach { line ->
+        addStatement(line)
+      }
+    }
+
     builder.addProperty(
       PropertySpec
         .builder(
           "descriptorMap",
-          Map::class.parameterizedBy(
-            String::class,
-            DescriptorProtos.FileDescriptorProto::class,
-          ),
+          descriptorMapClass,
         )
         .addModifiers(KModifier.PRIVATE)
         .initializer(
-          encoded.toList().fold(CodeBlock.builder().addStatement("mapOf(")) { b, (name, data) ->
+          descriptorMapInitializer.build(),
+        ).build(),
+    )
+  }
+
+  private fun subDescriptorMapCodeBlock(
+    builder: TypeSpec.Builder,
+    encodedList: List<Pair<String, String>>,
+    index: Int = 0,
```

Comment:
Just a minor comment

Is the default value needed? Since we always know the index, can we drop it `0`?

Reply:
Yeah, I'll do that